pkgs/development/tools/yarn2nix-moretea/yarn2nix/lib/generateNix.js

```diff
@@ -76,9 +76,14 @@ function fetchLockedDep(builtinFetchGit) {
       return ''
     }
 
-    const [url, sha1OrRev] = resolved.split('#')
+    const fileName = urlToName(resolved.split('#')[0])
+
+    if (resolved.startsWith('https://codeload.github.com/')) {
+      let s = resolved.split('/')
+      resolved = `git+https://github.com/${s[3]}/${s[4]}.git#${s[6]}`
```

Comment:
Rather than splitting again the resolved underneath, I would take care of that case completely inside this if statement and add a `return`.
I personally like trying to separate the cases clearly and not make them inter-related.